doc/introduction/measurements.rst

```diff
@@ -524,6 +524,80 @@ Collecting statistics for sequences of mid-circuit measurements is supported wit
     When collecting statistics for a list of mid-circuit measurements, values manipulated using
     arithmetic operators should not be used as this behaviour is not supported.
 
+Configuring mid-circuit measurements
+************************************
+
+As seen above, there are multiple ways in which circuits with mid-circuit measurements can be executed with
+PennyLane. For ease of use, we provide the following configuration options to users when initializing a
+:class:`~pennylane.QNode`:
+
+* ``mcm_method``: To set the method used for applying mid-circuit measurements. Use ``mcm_method="deferred"``
+  to use the deferred measurements principle or ``mcm_method="one-shot"`` to use the one-shot transform as
+  described above. When executing with finite shots, ``mcm_method="one-shot"`` will be the default, and
+  ``mcm_method="deferred"`` otherwise.
+
+  .. warning::
+
+      If the ``mcm_method`` argument is provided, the :func:`~pennylane.defer_measurements` or
+      :func:`~pennylane.dynamic_one_shot` transforms must not be applied directly to a :class:`~pennylane.QNode`
+      as it can lead to incorrect behaviour.
+
+* ``postselect_mode``: To configure how invalid shots are handled when postselecting mid-circuit measurements
+  with finite-shot circuits. Use ``postselect_mode="hw-like"`` to discard invalid samples. In this case, the number
+  of samples that are used for processing results will be less than or equal to the total number of shots. If
+  ``postselect_mode="fill-shots"`` is used, then the postselected value will be picked unconditionally, and all
+  samples will be considered valid. The default behaviour is ``postselect_mode="hw-like"``.
+
+  .. code-block:: python3
+  
+      import pennylane as qml
+      import numpy as np
+  
+      dev = qml.device("default.qubit", wires=3, shots=10)
+  
+      @qml.qnode(dev, mcm_method="one-shot", postselect_mode="fill-shots")
+      def circuit(x):
+          qml.RX(x, 0)
+          m0 = qml.measure(0, postselect=1)
+          qml.CNOT([0, 1])
+          return qml.sample(qml.PauliZ(0))
```

Comment:
Could we have an example comparing the two different modes? Something like:
```
def circuit(x):
     qml.RX(x, 0)
     m0 = qml.measure(0, postselect=1)
     qml.CNOT([0, 1])
     return qml.sample(qml.PauliZ(0))

fill_shots_qnode = qml.QNode(circuit, dev, mcm_method="one-shot", postselect_mode="fill_shots")
hw_like_qnode = qml.QNode(circuit, dev, mcm_method="one-shot", postselect_mode="hw-like")

fill_shots_qnode(np.pi / 2)
hw_like_qnode(np.pi / 2)
```

Might help clarify a lot of confusion.